pkg/frontend/adminactions/delete_managedresource.go

```diff
@@ -52,6 +53,14 @@ func (a *azureActions) ResourceDeleteAndWait(ctx context.Context, resourceID str
 		return a.deleteFrontendIPConfiguration(ctx, resourceID)
 	}
 
+	re = regexp.MustCompile(healthProbePattern)
+	// HealthProbes cannot be deleted with DeleteByIDAndWait either.
+	matches := re.FindStringSubmatch(resourceID)
+	if len(matches) > 0 {
+		loadBalancerName := matches[3]
+		return a.deleteHealthProbe(ctx, resourceID, loadBalancerName)
+	}
+
 	return a.resources.DeleteByIDAndWait(ctx, resourceID, apiVersion)
```

Comment:
I see 'resources' has been used before, but can we make more descriptive to on this file?